drivers/SmartThings/philips-hue/src/init.lua

```diff
@@ -1401,15 +1401,30 @@ cosock.spawn(function()
   end
 end, "Stray Hue Bulb Resolution Task")
 
+local function safe_wrap_handler(handler)
+  return function(driver, device, ...)
+    if device == nil or (device and device.id == nil) then
+      log.warn("Tried to handle capability command for device that has been deleted")
+      return
+    end
+    local success, result = pcall(handler, driver, device, ...)
+    if not success then
+      log.error_with({ hub_logs = true }, string.format("Failed to invoke capability command handler. Reason: %s", result))
+    end
+    return result
+  end
+end
+
 local disco = Discovery.discover
-local added = _initialize
-local init = _initialize
+local added = safe_wrap_handler(_initialize)
+local init = safe_wrap_handler(_initialize)
 
-local refresh_handler = handlers.refresh_handler
-local switch_on_handler = handlers.switch_on_handler
-local switch_off_handler = handlers.switch_off_handler
-local switch_level_handler = handlers.switch_level_handler
-local set_color_temp_handler = handlers.set_color_temp_handler
+local refresh_handler = safe_wrap_handler(handlers.refresh_handler)
+local switch_on_handler = safe_wrap_handler(handlers.switch_on_handler)
+local switch_off_handler = safe_wrap_handler(handlers.switch_off_handler)
+local switch_level_handler = safe_wrap_handler(handlers.switch_level_handler)
+local set_color_handler = safe_wrap_handler(handlers.set_color_handler)
+local set_color_temp_handler = safe_wrap_handler(handlers.set_color_temp_handler)
```

Comment:
Are these handlers not being run on the device thread meaning they are already being wrapped in a pcall?

Reply:
Ahh okay, I understand now why they are need